dwave/experimental/automorphism/automorphism_generation.py

```diff
@@ -0,0 +1,674 @@
+        (containing permutations other than the identity).
+        - u_vector: coset representatives with respect to base beta, grouped by
+        stabilizer index.
+        - neighbours: A precomputed dictionary of vertex neighbours for fast
+        membership.
+        - identity: A precomputed identity permutation for fast comparsion.
+        - vertex_block_index = A map of each vertex to which block of the
+        partition it is found in.
+        - best: The first permutation reached that yields the largest 
+        adjacency-matrix reading found so far.
+        - best_exist: Indicates whether any leaf node in the search tree has
+        been reached.
+        - beta: The current base, used as a reference permutation.
+    """
+    def __init__(
+            self,
+            graph: nx.Graph,
+            num_samples: int = 3,
+            seed: int = 42
+    ) -> None:
+        """
+        Initialize SchreierContext.
+    
+        Args:
+            graph: A NetworkX Graph object representing the input graph.
+            num_samples: Number of randomly sampled automorphisms to return.
+            seed: Seed used for reproducibility. Defaults to 42.
+        """        
+        self.graph: nx.Graph = graph
+        self.num_nodes: int = graph.number_of_nodes()
+        self.num_samples: int = num_samples
+        self.rng: random.Random = random.Random(seed)
+        self.leaf_nodes: int = 0
+        self.nodes_reached: int = 0
+        self.u_map: dict[int, int] = {}
+        self.u_len: int = 0
+        self.u_vector: list[list[NDArray[np.intp]]] = []
+        self.idx_to_node: dict[int, int] = {
+            idx: n for idx, n in enumerate(graph.nodes())
+        }
+        self.neighbours: dict[int, set[int]] = {
+            self.idx_to_node[i]: set(graph.neighbors(self.idx_to_node[i]))
+            for i in range(self.num_nodes)
+        }
+        self.identity: NDArray[np.intp] = np.array(
+            range(self.num_nodes),
+            dtype=np.intp
+        )
+        self.vertex_block_index: list[int] = [0] * self.num_nodes
+        self.vertex_block_index: dict[int, int] = {n: 0 for n in graph.nodes()}
+
+        self.best_perm: NDArray[np.intp] = np.array(
+            range(self.num_nodes),
+            dtype=np.intp
+        )
+        self.best_perm_exist: bool = False
+        self.beta: NDArray[np.intp] = np.array(
+            range(self.num_nodes),
+            dtype=np.intp)
+
+    def _sample_from_nested(self) -> list[NDArray[np.intp]]:
+        """Return a random sample of coset representatives.
+
+        Args:
+            ctx: A context object containing the following attributes:
+                - u_vector: coset representatives with respect to base beta, grouped 
+                by stabilizer index.
+                - num_samples: Number of randomly sampled automorphisms to return.
+                - rng: A random number generator instance used for sampling.
+        """
```

Comment:
No need to add `self` as a documented arg, but should have a return. Same for other methods below.
```suggestion
        """Generate a random sample of coset representatives.
        
        Returns:
            List of random samples of coset representatives.
        """
```